internal/deploy/values/builder.go

```diff
@@ -13,8 +13,9 @@ import (
 )
 
 type builder struct {
-	files  []string
-	values []map[string]interface{}
+	files                   []string
+	values                  []map[string]interface{}
+	dockerRegistryOverrides map[string]interface{}
```

Comment:
The bug that you discovered is the fact that value file overrides (--value-files) do not override default values, only the value overrides (-value) do. So it is applicable to `addGlobalDomainName` and `addGlobalTLSCrtAndKey`. Let's give this field a more generic name and also change above mentioned methods.
```suggestion
	defaultValues map[string]interface{}
```

Reply:
Let's also prefix all 3 methods with default to be consistent:
```go
addDefaultServerlessRegistryConfig
addDefaultGlobalDomainName
addDefaultGlobalTLSCrtAndKey
```